### DISCUSSION TITLE
Welcome to MPCC Discussions!

### DISCUSSION THREAD

**Discussion by alexliniger** [alexliniger/MPCC#55] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by coffear** [comment]
Hi, I would like to know if you tried a Simulink version of this algorithm, especially implemented with the built-in MPC toolbox in Simulink.
Thanks !

**Reply by alexliniger** [reply]
Maybe you can add the full C++ code into a simulink s-function block. But to be honest the last time I used Simulink is more than 10 years ago, so I am not sure what the best way is. 

Regarding the MPC toolbox, I assume it is quite some work. Maybe it is the easiest to use the nonlinear MPC approach and formulate the MPCC directly as a non linear problem. In this case the biggest issue is to add the track into the problem, which I don't know how to best do since I never worked with the toolbox.
